networkx/algorithms/community/modularity_max.py

```diff
@@ -234,7 +221,124 @@ def greedy_modularity_communities(G, weight=None, resolution=1, n_communities=1)
             b[v] += b[u]
             b[u] = 0
 
-    return sorted(communities.values(), key=len, reverse=True)
+        yield communities.values()
+
+
+def greedy_modularity_communities(
+    G, weight=None, resolution=1, cutoff=1, best_n=None, n_communities=None
+):
+    r"""Find communities in G using greedy modularity maximization.
+
+    This function uses Clauset-Newman-Moore greedy modularity maximization [2]_
+    to find the community partition with the largest modularity.
+
+    Greedy modularity maximization begins with each node in its own community
+    and repeatedly joins the pair of communities that lead to the largest
+    modularity until no futher increase in modularity is possible (a maximum).
+    Two keyword arguments adjust the stopping condition. `cutoff` is a lower
+    limit on the number of communities so you can stop the process before
+    reaching a maximum (used to save computation time). `best_n` is an upper
+    limit on the number of communities so you can make the process continue
+    until at most n communities remain even if the maximum modularity occurs
+    for more. To obtain exactly n communities, set both `cutoff` and `best_n` to n.
+
+    This function maximizes the generalized modularity, where `resolution`
+    is the resolution parameter, often expressed as $\gamma$.
+    See :func:`~networkx.algorithms.community.quality.modularity`.
+
+    Parameters
+    ----------
+    G : NetworkX graph
+
+    weight : string or None, optional (default=None)
+        The name of an edge attribute that holds the numerical value used
+        as a weight.  If None, then each edge has weight 1.
+        The degree is the sum of the edge weights adjacent to the node.
+
+    resolution : float, optional (default=1)
+        If resolution is less than 1, modularity favors larger communities.
+        Greater than 1 favors smaller communities.
+
```

Comment:
The `n_communities` parameter should still be listed here even though it's deprecated. The deprecation can be included in the docstring with the `.. deprecated` directive, e.g. something like:

```rst
n_communities : int

    .. deprecated:: 3.0
       The `n_communities` parameter is deprecated - use `cutoff` and/or
       `best_n` to set bounds on the desired number of communities instead.

    <original description for n_communities here>
```

Reply:
I used the old `n_communities` description as suggested, but isn't it clearer if we copy the `cutoff` docstring? It's supposed to replace `n_communities` and has the same functionality. But I'm fine either way.

Reply:
+1 for reusing the `cutoff` description instead if it's more clear

Reply:
done!